crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -705,22 +705,38 @@ impl CheckpointBuilder {
             );
 
         let span = error_span!("augment_epoch_last_checkpoint", tx_digest = ?cert.digest());
-        let signed_effect = self
+        let signed_effects = self
             .state
-            .try_execute_immediately(&cert, &self.epoch_store)
+            // Get the effects without committing execution to the db.
+            .get_change_epoch_tx_effects(&cert, &self.epoch_store)
             .instrument(span)
             .await?;
+
+        // We must write cert and effects to the state sync tables so that state sync is able to
+        // deliver to the transaction to CheckpointExecutor after it is included in a certified
+        // checkpoint.
+        let synced_txns = &self.state.database.perpetual_tables.synced_transactions;
```

Comment:
To be consistent with the behavior of commit certificate, I think the certificate should be written to the certificates table? Eventually they will become the same table but right now they are quite different. For instance, when a validator is asking other validators for the advance epoch tx, they will read it from the certificates table.

Reply:
Ok - the only reason I didn't do that was because I believe we always write certs and effects atomically, so I wasn't sure if anything would break if we only wrote a cert.

Reply:
Hmnm, writing to certs instead of sync_transactions causes test failures. Do you think its worth digging into?

Reply:
Specifically it seems to break reconfiguration

Reply:
That's interesting. I will need to break off that atomicity anyway as part of my next step on finalized transactions though (basically get rid of certs table). I will be back next Tue to continue that work.

Reply:
Ok, I found the cause, it was actually very simple. CheckpointExecutor assumes (quite reasonably) that synced_transactions is the only source of unexecuted transactions. I changed it to write to both and everything is fine. This isn't super clean, but I think we can clean this all up after the finalized transactions work is done.